src/lib/coda_base/pending_coinbase_intf.ml

```diff
@@ -0,0 +1,105 @@
+open Core_kernel
+open Import
+open Snark_params
+open Snarky
+open Tick
+open Fold_lib
+open Snark_bits
+
+module type S = sig
+  type t
+
+  module Coinbase : sig
+    type t
+
+    type var
+  end
+
+  module Index : sig
+    type t
+
+    val gen : t Quickcheck.Generator.t
+
+    module Vector : sig
+      type t = int
+
+      val length : t
+
+      val empty : t
+
+      val get : t -> t -> bool
+
+      val set : t -> t -> bool -> t
+    end
+
+    val to_bits : t -> bool list
+
+    val fold_bits : t -> bool Fold.t
+
+    val fold : t -> (bool * bool * bool) Fold.t
+
+    module Unpacked : sig
+      type var = Boolean.var sexp_list
+    end
+
+    include Bits_intf.S with type t := t
+  end
+
+  module Stack : sig
+    type t
+
+    type var
+  end
+
+  module Hash : sig
+    include Data_hash.Full_size
+
+    type path = Pedersen.Digest.t list
+
+    type _ Request.t +=
+      | Stack_path : Index.t -> path Request.t
+      | Get_coinbase_stack : Index.t -> (Stack.t * path) Request.t
+      | Set_coinbase_stack : Index.t * Stack.t -> unit Request.t
+      | Find_index_of_newest_stack : Index.t Request.t
+      | Find_index_of_oldest_stack : Index.t Request.t
+
+    val get : var -> Index.Unpacked.var -> (Coinbase.var, _) Checked.t
+
+    val merge : height:int -> t -> t -> t
+
+    val empty_hash : t
+
+    val of_digest : Pedersen.Digest.t -> t
+
+    val update_stack :
+         var
+      -> is_new_stack:Boolean.var
+      -> f:(   is_empty_or_writeable:Boolean.var
```

Comment:
what do these signify/why do we need them?

Reply:
We don't need them. I made this structure (similar to sparse ledger requests) before writing how this would be called. I've deleted it in the version I have now